contracts/Staking.sol

```diff
@@ -27,8 +31,10 @@ contract Staking {
 
     // a checkpoint of the valid balance of a user for an epoch
     struct Checkpoint {
-        uint256 epochId;
-        uint256 balance;
+        uint128 epochId;
+        uint16 multiplier;
+        uint256 startBalance;
```

Comment:
I would change this to `balance` because in the case of a `> newDeposits` withdrawal, it needs to change and it wouldn't be a "starting" balance.

I know this would mean some change in the code :(

Reply:
Nvm, we can change the meaning to start of epoch balance, works fine.